### ISSUE TITLE
Long pause or deadlock during Eclipse start up

### ISSUE THREAD

**Issue by sewe** [eclipse/buildship#592]
Testing the recent Eclipse Oxygen.1a RC2, I encountered a long pause or even a deadlock (I killed the IDE process after about two minutes) during start up, i.e., when the splash screen is shown.

As a suspicious number of `org.gradle.*` stack frames hold locks, I’ve filed the issue here. I hope the attached [jstack output](https://github.com/eclipse/buildship/files/1371307/jstack.txt) helps to analyze the problem (which I wasn’t able to reproduce for several IDE restarts so far).

**Comment by donat** [comment]
I've checked the jstack output and it seems it is from a wrong JVM. The stack frames show `CorePlugin.stop()` which means that it's from an Eclipse instance that is about to shut down. FTR a test was being executed when the shutdown happened.

So far I haven't seen this behavior, but I might have an idea. The Gradle classpath container can trigger project synchronization upon startup which might block the startup process. I'll try to check this behavior.

**Comment by sewe** [comment]
Yes, you are right. The `main` thread is also executing a shutdown (`EclipseStarter.shutdown`). Sorry for the confusion.

That being said, this definitely occurred during my Oxygen.1a RC2 test runs, as I normally don’t use Gradle and specifically imported [SpotBugs](https://github.com/spotbugs/spotbugs), which uses Gradle, and ran some of its tests.
